impl/src/main/java/com/okta/sdk/impl/resource/AbstractPropertyRetriever.java

```diff
@@ -279,7 +279,12 @@ protected Map getMapProperty(String key) {
 
         if (value != null) {
             if (value instanceof String) {
-                return enumConverter.fromValue(type, value.toString());
+                try {
```

Comment:
I believe we should do more that just log the error and proceed. We should potentially generate an "UNKNOWN" type for all enums in the project and categorize the undocumented values received from backend into this "UNKNOWN" type. 

More specifically, for https://github.com/okta/okta-sdk-java/issues/539#issue-807402912 we should categorize them as `Custom`

Reply:
There is 108 enumerations in the project right now. I am not sure that adding another value into each enum is optimal solution. May be it's better to patch only one enum (ApplicationSignOnMode) by adding "Custom" value as [Python SDK](https://github.com/okta/okta-sdk-python/pull/168/files#diff-4fcfe22de0cfed07e17d41396cdf3bdb6d16863389c26595b2c184166263f25cR41) did?

Reply:
Sounds reasonable to me.

Reply:
Also, please add a test that covers this use case.